### PR TITLE
added ROS2 method and added ROS2, ROS23 and ROS34PW methods to iipvsoop_test.jl

### PR DESCRIPTION
**Pull request by cwittens**
## Checklist

- [x] Appropriate tests were added
- [ ] Any code changes were done in a way that does not break public API
- [ ] All documentation related to code changes were updated
- [ ] The new code follows the
  [contributor guidelines](https://github.com/SciML/.github/blob/master/CONTRIBUTING.md), in particular the [SciML Style Guide](https://github.com/SciML/SciMLStyle) and
  [COLPRAC](https://github.com/SciML/COLPRAC).
- [ ] Any new documentation only uses public API
  
## Additional context

closes https://github.com/SciML/OrdinaryDiffEq.jl/issues/2112

### CODE REVIEW

File: src/generic_rosenbrock.jl [hunk 1]
```diff
@@ -872,6 +891,77 @@ end
...
+function ROS2Tableau() # 2nd order
+    gamma=1.7071067811865475 # 1+1/sqrt(2)
+    Alpha=[0     0;
+           1.    0]
+    Gamma=[gamma                   0;
+           -3.414213562373095   gamma]
+    B=[0.5, 0.5]
+    Bhat=[1, 0]
+    a,C,b,btilde,d,c=_transformtab(Alpha,Gamma,B,Bhat)
+    println(btilde)
```
Comment (ranocha) [inline src/generic_rosenbrock.jl:911]:
```suggestion
```
Debugging leftover?

File: src/generic_rosenbrock.jl [hunk 2]
```diff
@@ -872,6 +891,77 @@ end
...
+    RosenbrockAdaptiveTableau(a,C,b,btilde,gamma,d,c)
+end
+
+@doc """
+2nd order stiffly accurate Rosenbrock-Wanner method with 2 internal stages with (Rinf=0).
```
Comment (ranocha) [inline src/generic_rosenbrock.jl:916]:
```suggestion
    ROS2()

2nd order stiffly accurate Rosenbrock-Wanner method with 2 internal stages with (Rinf=0).
```

**Review by ranocha — CHANGES_REQUESTED** [review]
Thanks a lot! This looks mostly good to me. Did you test it also with some additional ODEs?

File: src/generic_rosenbrock.jl (hunk 1 above)
Comment (cwittens) [reply src/generic_rosenbrock.jl:911]: Ups, yes

**Comment by cwittens** [comment]
> Thanks a lot! This looks mostly good to me. Did you test it also with some additional ODEs?

What do you mean?

**Comment by ranocha** [comment]
> > Thanks a lot! This looks mostly good to me. Did you test it also with some additional ODEs?
> 
> What do you mean?

There are some tests in the `test` folder. Did you also run the new method for additional ODEs to check that they work as expected?

Anyway, nice work :+1:

**Review by ranocha — APPROVED** [review]

**Comment by ranocha** [comment]
Some Rosenbrock tests fail - I guess that's related to this PR?

**Comment by ChrisRackauckas** [comment]
yes definitely introduced here

**Comment by ranocha** [comment]
Most failing CI runs are caused by Codecov (see https://github.com/codecov/feedback/issues/301). The only exception is https://github.com/SciML/OrdinaryDiffEq.jl/actions/runs/8753631634/job/24023730348?pr=2165#step:6:956

Could you please check whether this PR causes that or whether this happens also on the current `master` branch?

**Comment by ChrisRackauckas** [comment]
This PR doesn't cause that. I need to fix that one up.

But those Rosenbrock tests need to be added here.

**Comment by ChrisRackauckas** [comment]
Okay yeah this looks good.

**Pull request merged**